### PR TITLE
EMP-610: updated mapping for idictable flag value

### PR DESCRIPTION
**Pull request by tanovellino**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java [hunk 1]
```diff
@@ -43,6 +44,13 @@ default String convertDecisionReason(CrmDecisionReasonDetailsModelInterface mode
                 model.getFull_grant_notes() : model.getReason_details();
     }
 
+    @Nullable
+    default Boolean convertIntegerToBoolean(Integer i) {
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java:48]: Do we need to mention the `default` keyword here?

Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java:48]: yes. it is the only way to declare a default function on an interface

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java [hunk 2]
```diff
@@ -43,6 +44,13 @@ default String convertDecisionReason(CrmDecisionReasonDetailsModelInterface mode
...
+        if (i == null) return null;
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java:49]: can we change to one liner as below?
`return i==null ? null : (i > 0);`

Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/CrmMapper.java:49]: I don't like 1 liners returning null or false as 1st option. It mislead to think that null is the most common when it is not

**Review by gpjustice — APPROVED** [review]

**Pull request merged**